### ISSUE TITLE
BigQuery: 'unparsable section' if `WITH OFFSET` is used without alias

### ISSUE THREAD

**Issue by VictorLeP** [sqlfluff/sqlfluff#1329]
## Expected Behaviour
BigQuery's `UNNEST` operator has an optional `WITH OFFSET` clause. Per the [BigQuery documentation](https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#unnest_operator), this clause has an _optional_ `AS` alias. The following query should therefore pass the linter:
```sql
SELECT i, offset
FROM UNNEST([1, 2, 3]) AS i WITH OFFSET
```

`offset` is the [implicit alias](https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#implicit_aliases).

## Observed Behaviour
The query fails to lint, with a PRS error:  "Found unparsable section: 'WITH OFFSET'".

The query lints fine when an alias is provided (support was added in https://github.com/sqlfluff/sqlfluff/issues/1009), so the following works:
```sql
SELECT i, offset
FROM UNNEST([1, 2, 3]) AS i WITH OFFSET AS offset
```

## Dialect
bigquery 

## Version
sqlfluff, version 0.6.4; Python 3.9.5
